CBinding/CMake/CMakeProject.cs

```diff
@@ -43,10 +43,53 @@ public class CMakeProject : FolderBasedProject
 		string name;
 		FilePath outputDirectory = new FilePath ("./bin");
 		CMakeFileFormat fileFormat;
+		CMakeToolchain cmakeToolchain;
+
+		/// <summary>
+		/// Occurs when a file is removed from this project.
+		/// </summary>
+		public event ProjectFileEventHandler FileRemovedFromProject;
+
+		/// <summary>
+		/// Occurs when a file is added to this project.
+		/// </summary>
+		public event ProjectFileEventHandler FileAddedToProject;
+
+		/// <summary>
+		/// Occurs when a file of this project has been modified
+		/// </summary>
+		public event ProjectFileEventHandler FileChangedInProject;
+
+		/// <summary>
+		/// Occurs when a file of this project has been renamed
+		/// </summary>
+		public event ProjectFileRenamedEventHandler FileRenamedInProject;
+
+
+		public bool HasLibClang { get; private set; }
+
+		public CLangManager ClangManager { get; private set; }
+
+		public SymbolDatabaseMediator DB { get; private set; }
+
+		public UnsavedFilesManager UnsavedFiles { get; private set; }
+
+		ProjectItemCollection items;
+		/// <summary>
+		/// Files of the project
+		/// </summary>
+		public ProjectFileCollection Files {
```

Comment:
I think this should be removed. Anything that uses it should instead access OnGetItemFiles (https://github.com/mhutch/cbinding/blob/master/CBinding/CMake/CMakeProject.cs#L220) which returns the actual files used by the CMake project.